internal/server/snapshot.go

```diff
@@ -0,0 +1,318 @@
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/resp"
+	"github.com/tidwall/tile38/internal/collection"
+	"github.com/tidwall/tile38/internal/log"
+)
+
+var errSnapshotLoadFailed = errors.New("snapshot load failed")
+
+const (
+	Id     = "id"
+	Offset = "offset"
+)
+
+// Record of the last snapshot for this dataset
+type SnapshotMeta struct {
+	path string
+
+	mu sync.RWMutex
+
+	_idstr	string
+	_offset	int64
```

Comment:
nit: not very conventional to use `_` as leading char of identifier in go

Reply:
Trying to follow the config code: https://github.com/housecanary/tile38/blob/master/internal/server/config.go#L47